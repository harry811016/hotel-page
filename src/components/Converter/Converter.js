import React from 'react'
import styled from '@emotion/styled'

const ConverterContainer = styled.div`
  display: flex;
  align-items: center;
  height: 1rem;
  p {
    margin-right: 10px;
  }
  select {
    border-radius: 5px;
    border: 1px solid #ffffff;
    padding: 0.25rem 0.5rem;
    color: #ffffff;
    background-color: rgba(255,105,15,0.5)
  }
`

const Converter = (props) => {
  const { handleCurrencyChange, currency } = props
  return (
    <ConverterContainer>
      <p>Currency</p>
      <select name="currency" id="currency" form="currencyForm" onChange={handleCurrencyChange} value={currency}>
        <option value="USD">USD</option>
        <option value="SGD">SGD</option>
        <option value="CNY">CNY</option>
        <option value="KRW">KRW</option>
      </select>
    </ConverterContainer>
  )
}

export default Converter