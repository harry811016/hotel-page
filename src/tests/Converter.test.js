import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import Converter from '../components/Converter/Converter'

const theme = {
  colors: {
    main_color: "#ff690f",
    black: "#000000",
    white: "#ffffff",
    dark_blue: "#003580",
    dark_yellow: "#feba02",
    dark_green: "#008009",
    dark_grey: "#333333",
    dark_red: "#bf323b",
    bg_green: "rgba(0,128,9,0.05)",
    bg_grey: "rgba(200, 200, 200, 0.5)"
  }
}

afterEach(cleanup)

it('should take a snapshot', () => {
  const { asFragment } = render(<ThemeProvider theme={theme}><Converter /></ThemeProvider>)
  expect(asFragment(<ThemeProvider theme={theme}><Converter /></ThemeProvider>)).toMatchSnapshot()
})

it('default currency is USD', () => {
  const { getByTestId } = render(<ThemeProvider theme={theme}><Converter /></ThemeProvider>)
  expect(getByTestId('currency')).toHaveTextContent('USD')
})

it('change currency to SGD', () => {
  const { getByTestId } = render(<ThemeProvider theme={theme}><Converter /></ThemeProvider>)
  fireEvent.click(getByTestId('sgd'))
  expect(getByTestId('currency')).toHaveTextContent('SGD')
})

it('fire handleCurrencyChange function when there is a changes of converter', () => {
  const handleCurrencyChange = jest.fn()
  const { getByTestId } = render(<ThemeProvider theme={theme}><Converter handleCurrencyChange={handleCurrencyChange} /></ThemeProvider>)
  fireEvent.change(getByTestId('currency'), { target: { value: 'CNY' } })
  expect(handleCurrencyChange).toHaveBeenCalled()
})