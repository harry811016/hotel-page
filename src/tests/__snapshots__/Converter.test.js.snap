// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1rm0f4a"
  >
    <p>
      Currency
    </p>
    <select
      data-testid="currency"
      form="currencyForm"
      id="currency"
      name="currency"
    >
      <option
        data-testid="usd"
        value="USD"
      >
        USD
      </option>
      <option
        data-testid="sgd"
        value="SGD"
      >
        SGD
      </option>
      <option
        data-testid="cny"
        value="CNY"
      >
        CNY
      </option>
      <option
        data-testid="krw"
        value="KRW"
      >
        KRW
      </option>
    </select>
  </div>
</DocumentFragment>
`;
